gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -857,6 +914,13 @@ protected void afterExecute(Runnable r, Throwable t) {
 
           // Compaction job of a dataset is successful.
           jobRunner.getDataset().setState(COMPACTION_COMPLETE);
+
+          // Add a new MPCompactor job if recompaction is necessary.
+          if (jobRunner.getDataset().needToRecompact()) {
```

Comment:
The recompaction job is based on a new `dataset`: 'recompactDataset', and its initial state is `unverified`.